### ISSUE TITLE
Credo 0.4.9 blows up with a list of excluded files.

### ISSUE THREAD

**Issue by austinsmorris** [rrrene/credo#160]
I have an umbrella app.  In my `/config/.credo.exs` file, I have defined some excluded files like so:

```
        included: ["lib/", "src/", "web/", "apps/"],
        excluded: [
          ~r"/_build/",
          ~r"/deps/",
          "apps/foo/mix.exs",
          "apps/foo/test/",
          "apps/bar/mix.exs",
          "apps/bar/test/",
          "apps/baz/mix.exs",
          "apps/baz/test/",
          "apps/bat/test/",
          "apps/bat/mix.exs",
        ]
```

When I run `mix credo` it blows up with the following error:

```
** (FunctionClauseError) no function clause matching in MapSet.difference/2
    (elixir) lib/map_set.ex:119: MapSet.difference(["apps/foo/config/dev.exs", "apps/foo/test/test_helper.exs", (blah blah blah long list of filenames], #MapSet<["./apps/foo/mix.exs"]>)
    lib/credo/sources.ex:53: Credo.Sources.exclude/2
    lib/credo/sources.ex:28: Credo.Sources.find/1
    lib/credo/cli/command/suggest.ex:41: anonymous fn/1 in Credo.CLI.Command.Suggest.load_and_validate_source_files/1
    (stdlib) timer.erl:166: :timer.tc/1
    lib/credo/cli/command/suggest.ex:39: Credo.CLI.Command.Suggest.load_and_validate_source_files/1
    lib/credo/cli/command/suggest.ex:16: Credo.CLI.Command.Suggest.run/2
    lib/credo/cli.ex:58: Credo.CLI.main/1
```

The problem looks to be in `Credo.Sources` here:

```
  defp exclude(files, []), do: files
  defp exclude(files, [pattern | remaining_patterns]) when is_binary(pattern) do
    exclude_paths = recurse_path(pattern) |> Enum.into(MapSet.new)

    files
    |> MapSet.difference(exclude_paths)
    |> exclude(remaining_patterns)
  end
```

`files` is a list, not a `MapSet`, so `MapSet.difference/2` gets mad...

**Comment by austinsmorris** [comment]
Sorry, I submitted before I sanitized the error and provided additional info.  The description should be good now...

**Comment by britto** [comment]
@austinsmorris 

1. Could you provide the exact value of your `excluded` key in your credo config? is there anything listed which is not a binary or a regex?
2. Are you running credo as a dependency from your mixfile or as standalone?

**Comment by britto** [comment]
@austinsmorris Nevermind! I could reproduce the bug locally, it happens when we have regexes **followed by** binaries in our exclusion patterns.

Thanks for reporting. I'm working on a fix 👍

For now, you can work around this bug by leaving all regexes at the end of the list.

**Comment by rrrene** [comment]
@austinsmorris Could try testing with `0.4.10-dev`?
@britto Could you review my patch: https://github.com/rrrene/credo/commit/aa475d237bef989cc8d4e5ce26fcf37bd2f48011

**Comment by austinsmorris** [comment]
@rrrene, have you pushed that branch/tag up yet?  I don't see it...

**Comment by rrrene** [comment]
It's a version on hex.pm: https://hex.pm/packages/credo

**Comment by rrrene** [comment]
Just use "0.4.10-dev" for `:credo` instead of say "~> 0.4" in your `mix.exs`.

**Comment by austinsmorris** [comment]
@rrrene, @britto,  got it!  It works!

**Comment by rrrene** [comment]
Just published `0.4.10`. Thx for reporting the bug!

**Comment by britto** [comment]
@rrrene Thanks! Sorry for the delay, got caught up with work last night.

Yes, that patch will fix the problem, however it breaks Elixir 1.1 compatibility. I'll post my solution soon in another PR.
